kv.rs

```diff
@@ -79,4 +81,96 @@ fn to_value_err(err: slog::Error) -> ValueError {
     }
 }
 
-// TODO: support going the other way
+/// Create a [`log::kv::Source`] for the key-value pairs for a slog record.
+pub(crate) fn get_kv_source<'a>(
+    record: &'a slog::Record<'a>,
+    logger_kv: &'a slog::OwnedKVList,
+) -> std::io::Result<Vec<(String, OwnedValue)>> {
```

Comment:
You've clearly spent some time figuring all this integration out (nice work by the way! Especially given the `log` key-value APIs are basically undocumented at this point), but if it's desirable to avoid this `Vec` I'd be keen to try figure out what forces it.

Reply:
Ah I see the lifetimes of keys and values simply don't play nicely between `slog`'s `Serializer` and `log`'s `Visitor`s. In `log`, we require an explicit lifetime for keys and values so that methods like `get()` can return borrowed data, so you aren't really left with much alternative. If this became a major issue we could consider alternative strategies to re-use allocations where possible and things like that.

Reply:
Yeah, the lifetimes don't quite match up. Ideally, slog would use a similar lifetime strategy. That might actually avoid needing allocations in some existing slog adapters.  But that would be a breaking change for slog, and probably wouldn't be worth the churn.